src/index.js

```diff
@@ -428,13 +432,16 @@ const acceptGrouping = backtrackOnFailure(state => {
 });
 
 const acceptDecimalEscape = backtrackOnFailure(state => {
-  let firstDecimal = state.eatAny(...decimalDigits.slice(1));
+  let firstDecimal = state.eatAny(...decimalDigits);
   if (firstDecimal === null) {
     return { matched: false };
   }
+  if (firstDecimal === '0') {
+    return { matched: true };
+  }
   // we also accept octal escapes here, but it is impossible to tell if it is a octal escape until all parsing is complete.
   // octal escapes are handled in acceptCharacterEscape for classes
-  state.backreferences.push(parseInt(firstDecimal + (state.eatNaturalNumber() || '')));
+  state.backreference(parseInt(firstDecimal + (state.eatNaturalNumber() || '')));
```

Comment:
It's not relevant to this PR, but can you add a test that 

```js
/\10(1)(2)(3)(4)(5)(6)(7)(8)(9)/u
```
is rejected and
```js
/\10(1)(2)(3)(4)(5)(6)(7)(8)(9)/
/\10(1)(2)(3)(4)(5)(6)(7)(8)(9)(10)/u
```
are accepted? I had to stare at this code for a while to convince myself it was doing the right thing, which means it would be good to have tests covering these cases.